### PR TITLE
Cleaned up the combined results JS and fixed save statuses and links …

### PR DESCRIPTION
**Pull request by EreMaijala**
…in combined results. Also fixed item status check when the same record was displayed in multiple columns, i.e. there can now be multiple elements in the elements array in both check_item_statuses.js and check_save_statuses.js in case the same record is displayed more than once.

This is continuation from #603.

### CODE REVIEW

File: themes/bootstrap3/js/combined-search.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/*global VuFind, setupOpenUrlLinks, checkSaveStatuses, setupSaveRecordLinks */
+VuFind.combinedSearch = (function() {
```
Comment (crhallberg) [inline themes/bootstrap3/js/combined-search.js]: I've been researching different object models, is this particular format worth using for the main VuFind object?

Comment (EreMaijala) [inline themes/bootstrap3/js/combined-search.js]: Well, that's of course always up to debate, but we have used it extensively when adding our own JS to VuFind, and it has worked marvelously, so I'd say "yes". I think it was modeled after the module pattern described e.g. here: https://toddmotto.com/mastering-the-module-pattern/.

Comment (demiankatz) [inline themes/bootstrap3/js/combined-search.js]: I like it! Though in this instance, there doesn't seem to be any reason for the (VuFind) at the end, as the closure you've defined doesn't take any parameters. Based on the article, it seems that one approach might be to actually attach the combinedSearch logic to the VuFind object *inside* the closure, which would make use of the VuFind parameter. Alternatively, you could keep it as it is, but omit the unused parameter at the end.

Comment (crhallberg) [inline themes/bootstrap3/js/combined-search.js]: I really like this model as well. After this branch is merged, I'll switch the VuFind namespace to this model for consistency. Sound good?

Comment (demiankatz) [inline themes/bootstrap3/js/combined-search.js]: Works for me. Aside from my nitpick about the (VuFind) parameter, anything we need to worry about before merging?

**Comment by demiankatz** [comment]
For reasons independent of this pull request, I decided to start building a Mink test for combined search functionality. This proved to be a worthwhile endeavor, as it very quickly revealed multiple bugs. I fixed some things in master, but one problem (failure to load item status information for AJAX-loaded columns) seemed easier to solve in this branch than in master. Thus, I fixed the problem here and added my test class to this branch. Clearly this is a sign that this work is a step in the right direction! I think we should be able to merge this soon.

**Comment by demiankatz** [comment]
@EreMaijala -- one question: your $.map callbacks return false in some circumstances. If I understand the documentation correctly, this may actually cause an element to be added to the returned array with a value of "false." Is this your intent, or is it better to return "null" and have the element excluded entirely? (Apologies if I'm just misunderstanding something here).

File: themes/bootstrap3/js/combined-search.js (hunk 1 above)
Comment (EreMaijala) [inline themes/bootstrap3/js/combined-search.js]: (VuFind) removed. There shouldn't be anything to worry, all this should work fine with the existing code, but of course I might have messed up something..

**Comment by EreMaijala** [comment]
@demiankatz, you're right. I just copied the $.map() logic from checkSaveStatuses(). I fixed it in both functions.

**Comment by demiankatz** [comment]
Thanks for this! I've manually merged to master, rewriting the commit history a bit for better granularity/clarity.

**Comment by demiankatz** [comment]
@EreMaijala: just a heads-up that I've also merged master into the (rather inaccurately named) lightbox-iframe branch (see #402) which I expect will itself be merged to master in a couple of days. Since this simplifies the lightbox code significantly, I was able to completely remove setupSaveRecordLinks() and all calls to it. I'm pretty sure this is fine, but since saving results from combined search is a local customization for you, I can't test exactly what you are doing. You may want to pay special attention to that scenario when testing the new branch.

**Comment by EreMaijala** [comment]
@demiankatz, thanks for the heads-up. It's not really that much of a local customization since the only thing you need to do to test it with bootprint3 out of the box is to take this out of combined-search.css:

```
.combined-list .result .right {
    display: none;
}
```

But I'll verify when it lands.

**Comment by demiankatz** [comment]
Thanks!

**Pull request closed without merging**